alicloud/errors.go

```diff
@@ -0,0 +1,50 @@
+package alicloud
+
+import (
+	"context"
+	"strings"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+)
+
+// isNotFoundError:: function which returns an ErrorPredicate for Alicloud API calls
+func isNotFoundError(notFoundErrors []string) plugin.ErrorPredicateWithContext {
+	return func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, err error) bool {
+		alicloudConfig := GetConfig(d.Connection)
+
+		// If the get or list hydrate functions have an overriding IgnoreConfig
+		// defined using the isNotFoundError function, then it should
+		// also check for errors in the "ignore_error_codes" config argument
+		allErrors := append(notFoundErrors, alicloudConfig.IgnoreErrorCodes...)
+		// Added to support regex in not found errors
+		for _, pattern := range allErrors {
+			if strings.Contains(err.Error(), pattern) {
+				return true
+			}
+		}
+		return false
+	}
+}
+
+// shouldIgnoreErrorPluginDefault:: Plugin level default function to ignore a set errors for hydrate functions based on "ignore_error_codes" config argument
+func shouldIgnoreErrorPluginDefault() plugin.ErrorPredicateWithContext {
+	return func(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, err error) bool {
+		if !hasIgnoredErrorCodes(d.Connection) {
+			return false
+		}
```

Comment:
Do you know if we need this block code?
I believe the following loop will address this check. If no ignore error code is configured in the .spc file, the loop will not be entered, and [the subsequent line of code](https://github.com/turbot/steampipe-plugin-alicloud/pull/344/files#diff-1502b5579feccb969e4e00de4e562b91392c75af3b00dc9e2e51359780f71896R43) will be executed, returning false.

```
for _, pattern := range alicloudConfig.IgnoreErrorCodes {
			if strings.Contains(err.Error(), pattern) {
				return true
			}
		}
```